bin/cron.js

```diff
@@ -109,7 +124,11 @@ cron.schedule("*/15 * * * *", async () => {
                     score += 100;
                     completed.practice += 1;
                     // If submission not already stored in MongoDB
-                    if (!userProgress.modules || !(moduleID in userProgress.modules)) {
+
+                    try {
+                      if (userProgress.modules.moduleID.practice) {
+                      }
+                    } catch {
```

Comment:
This needs to be refactored. This technically works in a very brute force and round-about way, but there are multiple issues with this: 
- Nesting a try-catch in a try-catch is unnecessary. The initial one is already going to prevent any uncaught errors, the second one may as well be an normal if statement 
- `if (userProgress.modules.moduleID.practice) { }` - using an empty if-statement in a try-catch is an extremely round-about way of asserting non-nullity. Having vital code dependent on an error being thrown to execute in a catch is bad practice, since often times, that's the actual code that needs to be within the try.
- `userProgress.modules.moduleID.practice` - This is **always** going to throw an error. You can't access the object of a given users module progress with `.moduleID`. It's going to try to access userProgress.modules's moduleID field, which doesn't exist. Then, `.practice` is trying to access undefined's practice attribute, which throws the error. You want something like `userProgress.modules[moduleID]` to access that.
- The reason why it seems to work is that since it always throws an error, it will always try to update mongo, instead of only updating the new changes.
- The solution to this is likely reverting the code back to the previous if-statement, then adding something like `if (!userProgress.modules || !(moduleID in userProgress.modules) || !userProgress.modules[moduleID].practice)`
Let me know if you have any questions.